cpp/src/prims/detail/nbr_intersection.cuh

```diff
@@ -788,10 +946,23 @@ nbr_intersection(raft::handle_t const& handle,
 
         local_nbrs_for_rx_majors.resize(
           local_nbr_offsets_for_rx_majors.back_element(handle.get_stream()), handle.get_stream());
+
+        optional_property_buffer_t optional_local_nbrs_properties{};
```

Comment:
Is this a buffer or view of a buffer (span)? Better distinguish the two by using two different type names consistently.